src/main/scala/com/microsoft/hyperspace/index/rules/IndexFilter.scala

```diff
@@ -90,6 +107,24 @@ trait IndexRankFilter extends IndexFilter {
    * @return Map of source plan to selected index
    */
   def apply(plan: LogicalPlan, applicableIndexes: PlanToIndexesMap): PlanToSelectedIndexMap
+
+  /**
+   * Set WHYNOT_REASON tag for unselected indexes.
+   *
+   * @param plan Plan to tag
+   * @param indexes Indexes to tag
+   * @param selectedIndex Selected index
+   */
+  protected def setRankReasonTag(
```

Comment:
function name sounds a bit awkward (sounds like you are setting something related to "rank")

Reply:
renamed `setUnselectedReasonTagForRank`